lib/pbench/server/__init__.py

```diff
@@ -236,25 +284,56 @@ def _get_valid_path(self, req_val, dir_val, logger):
                 if logger:
                     logger.error(
                         "The {} directory, does not resolve to a directory ('{}')",
-                        req_val,
+                        env_name,
                         dir_path,
                     )
                 return None
         return dir_path
 
-    def get_valid_dir_option(self, req_val, dir_val, logger):
-        """get_valid_dir_option - get the directory path raising an Error
-        if it is not a directory, returning a Path directory object.
+    def get_valid_dir_option(
+        self, env_name: str, dir_val: str, logger: Union[None, Logger]
+    ) -> Optional[Path]:
+        """Get the validated directory option from the given section.
+
+        Args:
+            env_name : Legacy environment name to display in error messages
+            dir_val : A directory value to resolve to a real directory
+            logger : Logger to use when low-level error messages should be
+                emitted
+
+        Returns:
+            A Path directory object, None if the directory value does not
+                resolve to a real directory.
         """
-        dir_path = self._get_valid_path(req_val, dir_val, logger)
+        dir_path = self._get_valid_path(env_name, dir_val, logger)
         if not dir_path:
             return None
 
         return dir_path
 
-    def timestamp(self):
-        """
-        "Return the current timestamp formatted as a string of the following form:
-                  <YYYY>-<MM>-<DD>T<hh>:<mm>:<ss>-<TZ>
+    def timestamp(self) -> str:
+        """Generate a time stamp string.
+
+        Returns:
+            The current timestamp formatted as a string of the following form:
+                <YYYY>-<MM>-<DD>T<hh>:<mm>:<ss>-<TZ>
         """
         return tstos(_time())
+
+
+def get_pbench_server_config(cfg_name: str) -> PbenchServerConfig:
+    """Construct a Pbench server configuration object and validate that all the
+    required properties are present.
+
+    Returns:
+        A PbenchServerConfig instance.
+    """
+    sc = PbenchServerConfig(cfg_name)
+
+    # The following with reference all the required properties tripping a raise
+    # of BadConfig if any of the properties are missing their base config value.
+    req_props = [getattr(sc, attr) for attr in PbenchServerConfig.REQ_PROPS]
+    # The following assertion will always be True, but it keeps linters quiet.
+    assert len(req_props) == len(PbenchServerConfig.REQ_PROPS)
+
+    return sc
```

Comment:
I went for number of characters changed in other files instead of purity here since I was trying to avoid proliferating changes.

It seems like everybody wants a class method for this, and I'd be happy to do that in a follow-on PR.

Reply:
Why not do it in this PR?  You would be making changes to all the same lines and modules that this PR is changing...putting it off to another PR would be needless churn to the `main` branch.

Reply:
Sorry, I am more and more against expanding scope of PRs.  We can iterate with follow-up PRs if somebody is that concerned about it.  What seems like small change to capture more ground can lead to unintended extra work.

Reply:
This request is not an expansion of scope.  It is a request to use a different idiom to solve the same problem, and the change of approach affects exactly the same lines.  And, I'm not the only one who has pointed out that the proffered approach isn't the most appropriate.

Reply:
I have tried this suggestion out, and it looked much better than what I had originally done.  Thanks for suggesting it!